metricflow/dataset/sql_dataset.py

```diff
@@ -52,7 +53,14 @@ def checked_sql_select_node(self) -> SqlSelectStatementNode:
         Otherwise, an exception is thrown.
         """
         if self._sql_select_node is None:
-            raise RuntimeError(f"{self} was created with a SQL node that is not a {SqlSelectStatementNode}")
+            raise RuntimeError(
+                str(
+                    LazyFormat(
+                        f"{self} was created with a SQL node that is not a {SqlSelectStatementNode}",
```

Comment:
Should we use the class name here instead of `self`? That way the logs will be more consistent/searchable and any changing attributes can go in the other args.
Also, wondering why `SqlSelectStatementNode` is in curly braces.

Reply:
I know this probably isn't an error you wrote - this doesn't need to block merge!

Reply:
Updated.